### PR TITLE
Introducing Lazy-decoding of page data in PageReader

### PR DESCRIPTION
**Pull request by lancelly**

### CODE REVIEW

File: java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java [hunk 1]
```diff
@@ -92,6 +94,38 @@ public AlignedPageReader(
     this.valueCount = valuePageReaderList.size();
   }
 
+  @TestOnly
```
Comment (JackieTien97) [inline java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java]: not only test use it, some class like `CompactionAlignedChunkReader` and `ReadChunkAlignedSeriesCompactionExecutor` will still use this constructor.

File: java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java [hunk 2]
```diff
@@ -67,7 +69,7 @@ public AlignedPageReader(
       ByteBuffer timePageData,
       Decoder timeDecoder,
       List<PageHeader> valuePageHeaderList,
-      List<ByteBuffer> valuePageDataList,
+      List<LazyLoadPageData> valuePageDataList,
```
Comment (JackieTien97) [inline java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java]: add UT for this new constructor

File: java/tsfile/pom.xml [hunk 3]
```diff
@@ -92,6 +92,12 @@
             <artifactId>gson</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>24.1.0</version>
+            <scope>compile</scope>
+        </dependency>
```
Comment (JackieTien97) [inline java/tsfile/pom.xml]:
```suggestion
```

File: java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java [hunk 4]
```diff
@@ -92,6 +94,38 @@ public AlignedPageReader(
...
+  public AlignedPageReader(
+      PageHeader timePageHeader,
+      ByteBuffer timePageData,
+      Decoder timeDecoder,
+      List<PageHeader> valuePageHeaderList,
+      List<ByteBuffer> valuePageDataList,
+      List<TSDataType> valueDataTypeList,
+      List<Decoder> valueDecoderList,
+      Filter globalTimeFilter,
+      boolean isLazyLoaded) {
```
Comment (JackieTien97) [inline java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java]: what's this parameter for? we shouldn't change the interface method signature

File: java/tsfile/src/test/java/org/apache/tsfile/read/reader/AlignedPageReaderPushDownTest.java [hunk 5]
```diff
@@ -137,7 +137,8 @@ private AlignedPageReader generateAlignedPageReader(
             valuePageDataList,
             valueDataTypeList,
             valueDecoderList,
-            globalTimeFilter);
+            globalTimeFilter,
+            false);
```
Comment (JackieTien97) [inline java/tsfile/src/test/java/org/apache/tsfile/read/reader/AlignedPageReaderPushDownTest.java]:
```suggestion
```

File: java/tsfile/src/test/java/org/apache/tsfile/read/reader/AlignedPageReaderPushDownTest.java [hunk 6]
```diff
@@ -163,7 +164,8 @@ private AlignedPageReader generateSingleColumnAlignedPageReader(
...
```
Comment (JackieTien97) [inline java/tsfile/src/test/java/org/apache/tsfile/read/reader/AlignedPageReaderPushDownTest.java]:
```suggestion
```

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java (hunk 1 above)
Comment (lancelly) [reply java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java]: Removed.

File: java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java (hunk 2 above)
Comment (lancelly) [reply java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java]: Done.

File: java/tsfile/pom.xml (hunk 3 above)
Comment (lancelly) [reply java/tsfile/pom.xml]: Done.

File: java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java (hunk 4 above)
Comment (lancelly) [reply java/tsfile/src/main/java/org/apache/tsfile/read/reader/page/AlignedPageReader.java]: I changed to LazyLoadPageData[].

File: java/tsfile/src/test/java/org/apache/tsfile/read/reader/AlignedPageReaderPushDownTest.java (hunk 5 above)
Comment (lancelly) [reply java/tsfile/src/test/java/org/apache/tsfile/read/reader/AlignedPageReaderPushDownTest.java]: Done.

File: java/tsfile/src/test/java/org/apache/tsfile/read/reader/AlignedPageReaderPushDownTest.java (hunk 6 above)
Comment (lancelly) [reply java/tsfile/src/test/java/org/apache/tsfile/read/reader/AlignedPageReaderPushDownTest.java]: Done.

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**